lib/auth/auth.go

```diff
@@ -906,9 +911,25 @@ func (a *Server) ExtendWebSession(user, prevSessionID, accessRequestID string, i
 		}
 	}
 
+	if req.SwitchBackDefaults != nil {
+		defaultRoles := req.SwitchBackDefaults.Roles
+		defaultExpiry := req.SwitchBackDefaults.Expires
+
+		if err := utils.StringSliceSubset(roles, defaultRoles); err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		if !defaultExpiry.IsZero() && defaultExpiry.Before(a.clock.Now().UTC()) {
+			return nil, trace.NotFound("web session has expired")
+		}
+
+		roles = req.SwitchBackDefaults.Roles
+		expiresAt = req.SwitchBackDefaults.Expires
```

Comment:
Does that have to take the current session's expiration into account? I don't have a full context on the switching back to defaults (is it what it is?), so I'm wondering whether it's possible that `request.SwitchBackDefaults.Expires > session.Expires`?